src/Razor/src/Microsoft.VisualStudio.Editor.Razor/DefaultVisualStudioRazorParser.cs

```diff
@@ -145,6 +145,25 @@ internal override Task<RazorCodeDocument> GetLatestCodeDocumentAsync(ITextSnapsh
             }
         }
 
+        // WebTools depends on this method. Do not remove until old editor is phased out
+        public override void QueueReparse()
+        {
+            // Can be called from any thread
+
+            if (_joinableTaskContext.IsOnMainThread)
+            {
+                ReparseOnUIThread();
+            }
+            else
+            {
+                _ = Task.Factory.StartNew(
+                    () => ReparseOnUIThread(),
+                    CancellationToken.None,
+                    TaskCreationOptions.None,
+                    TaskScheduler.FromCurrentSynchronizationContext());
```

Comment:
I think using `TaskScheduler.FromCurrentSynchronizationContext()` should work here because that's [what the old code did behind the scenes](https://github.com/dotnet/aspnetcore-tooling/blob/f81a59a1013b63d6625ba2dd950ebcda023cf491/src/Razor/src/Microsoft.VisualStudio.LanguageServices.Razor/VisualStudioForegroundDispatcher.cs#L17).

Reply:
So it's not entirely the same, the foreground dispatcher from the old would capture the "current synchronization context" on construction which would always be the UI thread. To be frank it was kind of a hack 😄. You'll need to use `_joinableTaskContext` to run this on the UI thread